Capstone/Assets/Game/Scripts/AIDirector.cs

```diff
@@ -0,0 +1,78 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Bladesmiths.Capstone
+{
+    public class AIDirector : MonoBehaviour
+    {
+        public GameObject[] enemyPrefabs;
+        public List<Enemy> enemyGroup;
+        public LinkedList<Enemy> attackQueue;
+        private static AIDirector instance;
+        private bool enemyAttacking;
+
+
+        public static AIDirector Instance { get { return instance; } }
+
+        private void Awake()
+        {
+            if (instance != null && instance != this)
+            {
+                Destroy(this.gameObject);
+            }
+            else
+            {
+                instance = this;
+                attackQueue = new LinkedList<Enemy>();
+                enemyGroup = new List<Enemy>();
+            }
+        }
+
+        private void Start()
+        {
+            
+        }
+
+        private void Update()
+        {
+            
+        }
+
+        public void AddToEnemyGroup(Enemy e)
+        {
+            if(e == null)
+            {
+                return;
+            }
+
+            enemyGroup.Add(e);
+
+        }
+
+        public void RemoveFromGroups(Enemy e)
+        {
+            if (!enemyGroup.Contains(e))
+            {
+                return;
+            }
+
+            enemyGroup.Remove(e);
```

Comment:
Remove returns a bool, by checking if it contains beforehand, you're actually running contains twice basically